astarte-utils/AstarteGenericProducer.cpp

```diff
@@ -129,14 +129,65 @@ bool AstarteGenericProducer::sendData(const QVariant &value, const QByteArray &t
 
 bool AstarteGenericProducer::sendData(const QList<QVariant> &value, const QByteArray &target, const QDateTime &timestamp, const QVariantHash &metadata)
 {
-    // TODO: checks
-    QHash<QByteArray, QByteArray> attributes;
-    attributes.insert("interfaceType", QByteArray::number(static_cast<int>(m_interfaceType)));
-    // TODO: handle reliability, retention and expiry
+    if (!target.startsWith('/') || target.endsWith('/') || target.contains("//")
```

Comment:
is all this code a duplicate of existing code? Can we reuse existing code / existing logic?